frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -208,6 +209,52 @@ COPY Dockerfile /bar
 	checkLinterWarnings(t, sb, dockerfile, []expectedLintWarning{})
 }
 
+func testUndeclaredArg(t *testing.T, sb integration.Sandbox) {
+	dockerfile := []byte(`
+ARG base=scratch
+FROM $base
+COPY Dockerfile .
+`)
+	checkLinterWarnings(t, sb, dockerfile, []expectedLintWarning{})
+
+	dockerfile = []byte(`
+ARG BUILDPLATFORM=linux/amd64
+FROM --platform=$BUILDPLATFORM scratch
```

Comment:
This case should work even without the `ARG` definition as `BUILDPLATFORM` is automatically defined.

You can leave a case with custom arg like you had before and then this case with implicit arg.

For the typo case, no `ARG` definition.